src/coreclr/vm/codeman.cpp

```diff
@@ -1336,27 +1336,6 @@ void EEJitManager::SetCpuInfo()
 
     CORJIT_FLAGS CPUCompileFlags;
 
-#if defined(TARGET_X86)
-    CORINFO_CPU cpuInfo;
-    GetSpecificCpuInfo(&cpuInfo);
-
-    switch (CPU_X86_FAMILY(cpuInfo.dwCPUType))
-    {
-        case CPU_X86_PENTIUM_4:
-            CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_TARGET_P4);
-            break;
-
-        default:
-            break;
-    }
-
-    if (CPU_X86_USE_CMOV(cpuInfo.dwFeatures))
-    {
-        CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_USE_CMOV);
-        CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_USE_FCOMI);
-    }
```

Comment:
For `cmov`/`fcomi` do we want to keep the CPUID check as part of the below path and fail to start if missing like we do with SSE/SSE2?
```
EEPOLICY_HANDLE_FATAL_ERROR_WITH_MESSAGE(COR_E_EXECUTIONENGINE, W("CMOV is not supported on the processor."));
```

The bits are in `EDX` and `15` (`CMOV`) and `0` (`x87 FPU`, only needed for 32-bit). `FCOMI` support is indicated by the presence of both `CMOV` and `x87 FPU`).

It's highly unlikely anyone tries to run on such an old system and I believe even the Linux kernel requires `CMOV` (and `x87 FPU` for 32-bit), but it might be good for consistency

Reply:
-- I don't think we need a proper `InstructionSet_*` for these, as they are just part of our `X86Base`, much like `CpuId` is.

Reply:
I would be ok with adding debug-only asserts for these as a form of documentation. I do not think we need to have error messages for these.

The error message for SSE2 is unreachable code, on Windows at least. I am pretty sure we would crash much earlier without SSE2 support. Windows toolset requires SSE2 since 2011. I know that because we got a spike of .NET Framework crash reports very early during startup. The spike was tracked down to C++ toolset update that came with SSE2 minbar. The problem was won't fixed - there was not enough machines without SSE2 even back then.

Reply:
Sounds good.

Just noting that I believe Linux still supports no SSE/SSE2 for 32-bit, so it may be hittable there.